theseus_gui/src/components/ui/JavaSelector.vue

```diff
@@ -85,8 +90,14 @@ const testingJavaSuccess = ref(null)
 async function testJava() {
   testingJava.value = true
   let result = await get_jre(props.modelValue ? props.modelValue.path : '')
+
   testingJava.value = false
-  testingJavaSuccess.value = !!result
+  if (result) {
+    let [majorVersion, minorVersion] = await extract_version_from_string(result.version)
+    testingJavaSuccess.value = majorVersion == 1 && minorVersion == props.version
+  } else {
+    testingJavaSuccess.value = false
+  }
```

Comment:
this should be moved to the actual testing function, and not in javascript imo. the testing function should be able to get the java version and you can validate it.